jupyros/ros_widgets.py

```diff
@@ -287,4 +287,41 @@ def step_clk(arg):
     btm_box = widgets.VBox([bgpath_box, options_hbox, duration_box, start_box, que_box, factor_box, delay_box, buttons_hbox, out_box])
     widget_list.append(btm_box)
     vbox = widgets.VBox(children=widget_list)
+    return vbox
+
+
+def client(srv_name, srv_type):
+    """
+    Create a form widget for message type srv_type.
+    This function analyzes the fields of srv_type and creates
+    an appropriate widget.
+
+    @param srv_type The service message type
+    @param srv_name The service name to call
+
+    @return jupyter widget for display
+    """
+
+    widget_list = []
+    widget_dict = {}
+
+    add_widgets(srv_type._request_class(), widget_dict, widget_list)
+    call_btn = widgets.Button(description="Call Service")
+
+    def call_srv(arg):
+        msg_to_send = srv_type._request_class()
+        widget_dict_to_msg(msg_to_send, widget_dict)
+        rospy.wait_for_service(srv_name)
```

Comment:
we could move the wait_for_service call outside the callback, and we could also give a timeout (it's an optional parameter). 

What do you think?

Reply:
Is it necessary to wait for the service? We could also simply check if the service is there and throw an error if it's not, would that be better?